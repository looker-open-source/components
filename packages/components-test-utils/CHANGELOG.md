# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
