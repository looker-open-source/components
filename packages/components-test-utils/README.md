# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
