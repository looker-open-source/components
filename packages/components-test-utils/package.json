{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
