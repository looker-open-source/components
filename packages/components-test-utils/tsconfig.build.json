{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
