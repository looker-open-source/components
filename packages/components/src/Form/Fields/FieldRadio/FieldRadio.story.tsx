/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React from 'react'
import { Story } from '@storybook/react/types-6-0'
import { FieldRadio, FieldRadioProps } from './FieldRadio'

const Template: Story<FieldRadioProps> = (args) => <FieldRadio {...args} />

export const Basic = Template.bind({})
Basic.args = {
  id: 'fieldRadioId',
  label: 'Field Radio Example',
  name: 'thumbsUp',
}

export const Detail = Template.bind({})
Detail.args = {
  ...Basic.args,
  detail: 'Some exciting details that describe the use of this.',
}

export const Checked = Template.bind({})
Checked.args = {
  ...Basic.args,
  checked: true,
}

export const Disabled = Template.bind({})
Disabled.args = {
  ...Basic.args,
  disabled: true,
}

export const DisabledChecked = Template.bind({})
DisabledChecked.args = {
  ...Checked.args,
  disabled: true,
}

export const ReadOnly = Template.bind({})
ReadOnly.args = {
  ...Basic.args,
  readOnly: true,
}

export default {
  component: FieldRadio,
  title: 'FieldRadio',
}
