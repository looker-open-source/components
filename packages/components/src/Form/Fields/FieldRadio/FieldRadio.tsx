/*

 MIT License

 Copyright (c) 2019 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import React, { forwardRef, Ref } from 'react'
import styled from 'styled-components'
import uuid from 'uuid/v4'
import { useFormContext } from '../../Form'
import { Radio, RadioProps } from '../../Inputs/Radio/Radio'
import { Field, FieldProps, omitFieldProps, pickFieldProps } from '../Field'

export interface FieldRadioProps
  extends RadioProps,
    Omit<FieldProps, 'description' | 'detail'> {}

const FieldRadioComponent = forwardRef(
  (props: FieldRadioProps, ref: Ref<HTMLInputElement>) => {
    const validationMessage = useFormContext(props)
    const { id = uuid() } = props
    return (
      <Field
        id={id}
        alignLabel="right"
        alignValidationMessage="right"
        validationMessage={validationMessage}
        {...pickFieldProps(props)}
      >
        <Radio
          {...omitFieldProps(props)}
          id={id}
          validationType={validationMessage && validationMessage.type}
          ref={ref}
        />
      </Field>
    )
  }
)

FieldRadioComponent.displayName = 'FieldRadioComponent'

export const FieldRadio = styled(FieldRadioComponent)``
