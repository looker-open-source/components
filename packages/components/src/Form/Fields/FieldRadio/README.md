# FieldRadio

The `<FieldRadio />` component is composed of a `<Radio />` component and a `<Label />` component. Using `<FieldRadio />` allows for rendering validation messages. By default, the label will render to the right of the radio button, however this is adjustable with the `inline` property.

```tsx
function Basic() {
  return (
    <FieldRadio id="fieldRadioId" label="Field Radio Example" name="thumbsUp" />
  );
}
```
