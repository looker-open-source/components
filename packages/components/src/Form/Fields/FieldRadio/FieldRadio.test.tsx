/*

 MIT License

 Copyright (c) 2021 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import 'jest-styled-components'
import React from 'react'
import { renderWithTheme } from '@looker/components-test-utils'
import { screen } from '@testing-library/react'
import { FieldRadio } from './FieldRadio'

describe('FieldRadio', () => {
  test('FieldRadio renders with label', () => {
    renderWithTheme(<FieldRadio id="FieldRadioID" label="FM" name="thumbsUp" />)

    expect(screen.getByLabelText('FM')).toBeInTheDocument()
  })

  test('FieldRadio renders description and detail props', () => {
    const { getByText } = renderWithTheme(
      <FieldRadio
        description="describe something here."
        detail="0/50"
        id="FieldRadioID"
        label="FM"
        name="thumbsUp"
      />
    )
    expect(getByText('describe something here.')).toBeInTheDocument()
    expect(getByText('0/50')).toBeInTheDocument()
  })

  test('FieldRadio renders disabled', () => {
    const { getByRole } = renderWithTheme(
      <FieldRadio disabled id="FieldRadioID" label="FM" name="thumbsUp" />
    )
    const RadioInput = getByRole('radio')

    expect(RadioInput as HTMLInputElement).toBeDisabled()
  })

  test('FieldRadio renders with error message', () => {
    const errorMessage = 'This is an error'

    const { getByText } = renderWithTheme(
      <FieldRadio
        id="FieldRadioID"
        label="FM"
        name="thumbsUp"
        validationMessage={{ message: errorMessage, type: 'error' }}
      />
    )

    expect(getByText('This is an error')).toBeInTheDocument()
  })
})
