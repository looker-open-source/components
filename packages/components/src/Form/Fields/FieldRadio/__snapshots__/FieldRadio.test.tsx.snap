// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A FieldRadio 1`] = `
.c6 {
  position: relative;
  width: 100%;
  height: 100%;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-radius: 50%;
  border-color: currentColor;
  color: transparent;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear, box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear, box-shadow 25ms linear;
}

.c6::after {
  content: '';
  position: absolute;
  background: #fff;
  width: 6px;
  height: 6px;
  border-radius: 50%;
  top: calc(50% - 3px);
  right: calc(50% - 3px);
}

.c4 {
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c4 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c4 input:checked + .c5 {
  color: #6C43E0;
}

.c4 input:not(:checked) + .c5 {
  border-color: #DEE1E5;
  background: #FFFFFF;
}

.c4 input:focus + .c5 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c4 input:disabled + .c5 {
  color: #939BA5;
}

.c4 input:disabled:not(:checked) + .c5 {
  background: transparent;
}

.c4 input:disabled:not(:checked) + .c5::after {
  background: transparent;
}

.c2 {
  color: #343C42;
  font-size: 0.75rem;
  font-weight: 600;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-template-areas: 'input label' '. messages';
  grid-template-columns: repeat(3,max-content);
  line-height: 1.25rem;
}

.c0 .c3 {
  grid-area: input;
}

.c0 .c1 {
  font-size: 0.875rem;
  font-weight: normal;
  grid-area: label;
  padding-left: 0.5rem;
}

.c0 .c7 {
  grid-area: messages;
  padding-left: 0.75rem;
}

<label
  className="c0 "
  htmlFor="FieldRadioID"
>
  <span
    className="c1 c2"
    color="text2"
    fontSize="xsmall"
    fontWeight="semiBold"
  >
    üëç
  </span>
  <div
    className="c3 "
  >
    <div
      className="c4 "
    >
      <input
        aria-describedby="FieldRadioID-describedby"
        id="FieldRadioID"
        name="thumbsUp"
        type="radio"
      />
      <div
        className="c5 c6"
      />
    </div>
  </div>
  <div
    className="c7 "
    id="FieldRadioID-describedby"
  />
</label>
`;

exports[`A FieldRadio checked 1`] = `
.c6 {
  position: relative;
  width: 100%;
  height: 100%;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-radius: 50%;
  border-color: currentColor;
  color: transparent;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear, box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear, box-shadow 25ms linear;
}

.c6::after {
  content: '';
  position: absolute;
  background: #fff;
  width: 6px;
  height: 6px;
  border-radius: 50%;
  top: calc(50% - 3px);
  right: calc(50% - 3px);
}

.c4 {
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c4 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c4 input:checked + .c5 {
  color: #6C43E0;
}

.c4 input:not(:checked) + .c5 {
  border-color: #DEE1E5;
  background: #FFFFFF;
}

.c4 input:focus + .c5 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c4 input:disabled + .c5 {
  color: #939BA5;
}

.c4 input:disabled:not(:checked) + .c5 {
  background: transparent;
}

.c4 input:disabled:not(:checked) + .c5::after {
  background: transparent;
}

.c2 {
  color: #343C42;
  font-size: 0.75rem;
  font-weight: 600;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-template-areas: 'input label' '. messages';
  grid-template-columns: repeat(3,max-content);
  line-height: 1.25rem;
}

.c0 .c3 {
  grid-area: input;
}

.c0 .c1 {
  font-size: 0.875rem;
  font-weight: normal;
  grid-area: label;
  padding-left: 0.5rem;
}

.c0 .c7 {
  grid-area: messages;
  padding-left: 0.75rem;
}

<label
  className="c0 "
  htmlFor="FieldRadioID"
>
  <span
    className="c1 c2"
    color="text2"
    fontSize="xsmall"
    fontWeight="semiBold"
  >
    üëç
  </span>
  <div
    className="c3 "
  >
    <div
      className="c4 "
    >
      <input
        aria-describedby="FieldRadioID-describedby"
        checked={true}
        id="FieldRadioID"
        name="thumbsUp"
        type="radio"
      />
      <div
        className="c5 c6"
      />
    </div>
  </div>
  <div
    className="c7 "
    id="FieldRadioID-describedby"
  />
</label>
`;
